packages/p2p/src/components/modal-manager/modals/ad-created-modal/ad-created-modal.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { Button, Checkbox, Modal, Text } from '@deriv/components';
+import { isMobile } from '@deriv/shared';
+import { observer } from '@deriv/stores';
+import { localize, Localize } from 'Components/i18next';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import { api_error_codes } from 'Constants/api-error-codes';
+import { useStores } from 'Stores';
+
+const AdCreatedModal = () => {
+    const { my_ads_store } = useStores();
+    const { adverts_archive_period, advert_details, setIsAdCreatedModalVisible, setShowAdForm } = my_ads_store;
+    const { hideModal, is_modal_open, showModal } = useModalManagerContext();
+    const should_not_show_auto_archive_message_again = React.useRef(false);
+    const { AD_EXCEEDS_BALANCE, AD_EXCEEDS_DAILY_LIMIT } = api_error_codes;
+
+    const onCheckboxChange = () =>
+        (should_not_show_auto_archive_message_again.current = !should_not_show_auto_archive_message_again.current);
+
+    const onClickOkCreatedAd = () => {
+        localStorage.setItem(
+            'should_not_show_auto_archive_message',
+            JSON.stringify(should_not_show_auto_archive_message_again.current)
+        );
+        setIsAdCreatedModalVisible(false);
+        hideModal();
+
+        //TODO: refactor below section - use advert_details directly from AdVisibilityErrorModal once it's refactored
+        if (advert_details?.visibility_status?.includes(AD_EXCEEDS_BALANCE)) {
+            showModal(
+                {
+                    key: 'AdVisibilityErrorModal',
+                    props: { error_code: AD_EXCEEDS_BALANCE },
+                },
+                { should_stack_modal: isMobile() }
+            );
+        } else if (advert_details?.visibility_status?.includes(AD_EXCEEDS_DAILY_LIMIT)) {
+            showModal(
+                {
+                    key: 'AdVisibilityErrorModal',
+                    props: { error_code: AD_EXCEEDS_DAILY_LIMIT },
+                },
+                { should_stack_modal: isMobile() }
+            );
```

Comment:
Maybe something like:
```js
if ([AD_EXCEEDS_BALANCE, AD_EXCEEDS_DAILY_LIMIT].includes(advert_details?.visibility_status) {
            showModal(
                {
                    key: 'AdVisibilityErrorModal',
                    props: { error_code: advert_details.visibility_status }
                },
                { should_stack_modal: isMobile() }
            );
}

```

Reply:
advert_details.visibility_status is an array. so i would not be able to pass the specific status. btw the error checking will be removed from this section once advisibilityerrormodal refactoring is done. i have added a todo comment above.